salutations.php

```diff
@@ -303,18 +327,22 @@ function salutation_update($type, $type_field_id, $contact_id, $salutation_optio
     'greetingString' => $salutation_string['values'][0]['label']
   ]);
 
-  //Update the eisting saluation
-  $custom_group_id = civicrm_api3('CustomGroup', 'getvalue', [
-    'return' => "id",
-    'name' => "salutations",
-  ]);
+  //Update the existing saluation
+  static $custom_group_id;
+  if (!$custom_group_id) {
+    $custom_group_id = civicrm_api3('CustomGroup', 'getvalue', [
+      'return' => "id",
+      'name' => "salutations",
+    ]);
+  }
   $custom_group_table = "civicrm_value_salutations_" . $custom_group_id ."_id";
-  $salutation_id = $salutation_option_selected['values'][0][$custom_group_table];
   $salutation = $processed_salutation['values']['greeting'];
-  $contactSalutationUpdate = civicrm_api3('CustomValue', 'create', [
-    'entity_id' => $contact_id,
-    "custom_salutations:salutation:$salutation_id" => "$salutation",
-  ]);
+  if ($salutation != $current_calculated_salutation) {
```

Comment:
I pass through the current value of the calculated salutation so I can skip making a database write when the value matches.